zosbuild/Makefile

```diff
@@ -0,0 +1,312 @@
+all: check protoc
```

Comment:
Sorry but I don't think we can take on another build system. We have too many build systems for C++ already (autotools, CMake, and Bazel). What prevents the autotools and CMake builds from working with zOS?

Reply:
Actually zOS do not have cmake, configure and tools like automake and autotools. But if that's the case I can remove makefile from the PR.

Reply:
Also the compiler on zOS is different and needed different FLAGS. We faced with numerous compile and link problems using configure and cmake. But since make is the most basic option and also available on zos for long time and is stable we turned to use that and we were able to compile the protobuf.

Reply:
Are you saying that CMake will work? Actually the source tar.gz files we publish at https://github.com/protocolbuffers/protobuf/releases also include a `configure` script, so you should be able to run `./configure && make` without even needing autotools.

Reply:
Actually CMake is not available on zOS at all. Configure also does not work as intended. The only way for mainframe users to compile the project is using pure make. That's way an old fashioned makefile was created.

Reply:
in order to have configure, first there is a need to run `autogen.sh` which failed on zOS machine since it has dependencies not available or fully functional(for instance `autoreconf`) on zOS as well.

Reply:
There's no need to run autogen.sh or autoreconf if you start with one of our packed tar.gz files, which already contains the configure script.